### ISSUE TITLE
[FR] fread could use 'strip.white' argument

### ISSUE THREAD

**Issue by nigmastar** [Rdatatable/data.table#1387]
I noticed that `fread` removes all leading and trailing spaces when reading in character mode (colClasses). The code below reproduces the error:

```
library(data.table)

sessionInfo()
# R version 3.2.2 (2015-08-14)
# Platform: x86_64-w64-mingw32/x64 (64-bit)
# Running under: Windows 7 x64 (build 7601) Service Pack 1
# 
# locale:
#     [1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
# [3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
# [5] LC_TIME=English_United Kingdom.1252    
# 
# attached base packages:
#     [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#     [1] data.table_1.9.6
# 
# loaded via a namespace (and not attached):
#     [1] tools_3.2.2  chron_2.3-47

lines <- c("S,,  52              , ,6262276",
           "F,,53, ,6245264",
           "G,,54, ,6733283")

writeLines(lines, "fread_trailing_sample.csv")

dt <- fread("fread_trailing_sample.csv", colClasses="character")
dt$V3
# [1] "52" "53" "54"

df <- read.table("fread_trailing_sample.csv", sep=",", colClasses="character")

df$V3
# [1] "  52              " "53"                 "54"
```

**Comment by arunsrinivasan** [comment]
Yes, `fread()` strips white spaces by default, unless the field is quoted. What's the *error* here? And what's the use case of having the spaces unstripped?

**Comment by nigmastar** [comment]
Hi,

I think a reader, when asked not to guess the data type, should never
modify the raw data (including whitespaces).

When working with client data, loading the data 'as is' is crucial.

**Comment by arunsrinivasan** [comment]
I don't think of removing leading/trailing whitespace as "modifying raw data". But a `strip.white` argument should be straightforward to add with default `TRUE` (since `fread()` aims at keeping meaningful defaults).

**Comment by arunsrinivasan** [comment]
Oops.. it's already there. You just have to do:

```r
dt <- fread("fread_trailing_sample.csv", colClasses="character", strip.white=FALSE)
#    V1 V2                 V3 V4      V5
# 1:  S      52                  6262276
# 2:  F                    53    6245264
# 3:  G                    54    6733283
```

**Comment by arunsrinivasan** [comment]
Implemented in commit 0e7a83572.
